File: frontend/types/document-request.ts
/**
 * document-request.ts — Student document-request types.
 *
 * TODO: sync with backend when schema is implemented.
 *       backend/app/schemas/document_request.py does not exist yet
 *       (ROADMAP 4.8). The SQLAlchemy model `DocumentRequest` is the only
 *       source right now — these shapes match its columns.
 */

import type {
  DocumentStatus,
  DocumentType,
  IsoDate,
  IsoDateTime,
  Uuid,
} from "./common"

export interface DocumentRequestCreate {
  document_type: DocumentType
  note?: string | null
}

export interface DocumentRequestResponse {
  id: Uuid
  student_id: Uuid
  document_type: DocumentType
  note: string | null
  status: DocumentStatus
  admin_note: string | null
  pickup_date: IsoDate | null
  processed_by: Uuid | null
  created_at: IsoDateTime
  updated_at: IsoDateTime
}

export interface DocumentRequestApprove {
  pickup_date: IsoDate
  admin_note?: string | null
}

export interface DocumentRequestReject {
  admin_note: string
}

File: backend/app/schemas/document_request.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.models.enums import DocumentStatus, DocumentType


class DocumentRequestCreate(BaseModel):
    document_type: DocumentType
    note: str | None = Field(default=None, max_length=2000)


class DocumentRequestApproveRequest(BaseModel):
    pickup_date: date
    admin_note: str | None = Field(default=None, max_length=2000)


class DocumentRequestRejectRequest(BaseModel):
    admin_note: str = Field(min_length=1, max_length=2000)


class DocumentRequestResponse(BaseModel):
    id: UUID
    student_id: UUID
    document_type: DocumentType
    note: str | None
    status: DocumentStatus
    admin_note: str | None
    pickup_date: date | None
    processed_by: UUID | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
